src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+                }
+            });
+        }
+
+        [ConditionalFact(nameof(IsAdmin_IsNotNano_RemoteExecutorIsSupported))] // Nano has no "netapi32.dll", Admin rights are required
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [OuterLoop("Requires admin privileges")]
+        public void TestUserNetworkCredentialsPropertiesOnWindows()
+        {
+            string testFilePath = GetTestFilePath();
+            Assert.False(string.IsNullOrWhiteSpace(testFilePath), $"Path to test file should not be empty: {testFilePath}");
+
+            string testFilePathRoot = Path.GetPathRoot(testFilePath);
+            const string ShareName = "testForDotNet";
+            using TestFileShare fileShare = new TestFileShare(ShareName, Path.GetDirectoryName(testFilePath));
+            string testFileUncPath = $"\\\\localhost\\{ShareName}\\{Path.GetFileName(testFilePath)}";
+            string testFileContent = Guid.NewGuid().ToString();
+            File.WriteAllText(testFilePath, testFileContent);
+
+            const string LocalPath = nameof(LocalPath);
+            const string UncPath = nameof(UncPath);
+            using Process p = CreateProcess(() =>
+            {
+                // Read file content using network credentials and output it for assertion comparison
+                Console.Write(File.ReadAllText(Environment.GetEnvironmentVariable(UncPath)));
+                Console.Write(ItemSeparator);
+
+                try
+                {
+                    // Read file content using current user - should fail, because impersonated user
+                    // isn't explicitly authorized to access the file.
+                    _ = File.ReadAllText(Environment.GetEnvironmentVariable(LocalPath));
+                    Console.Write('1');
+                }
+                catch (SecurityException)
+                {
+                    Console.Write('0');
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    Console.Write('0');
+                }
+
+                return RemoteExecutor.SuccessExitCode;
+            });
+            p.StartInfo.Environment[LocalPath] = testFilePath;
+            p.StartInfo.Environment[UncPath] = testFileUncPath;
+            p.StartInfo.StandardOutputEncoding = Encoding.UTF8;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.UseCredentialsForNetworkingOnly = true;
+
+            Action<string, string> setup = (username, _) =>
+            {
+                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
+                {
+                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: true);
+                }
+            };
```

Comment:
nit: this is very subjective, but you can use local C# functions if you find them easier to use/more readable. We typically put them at the end of the method.

Moreover, you are passing `username` but capturing `testFilePath`, so you can just capture all locals and pass an `Action` instead of `Action<string, string>` to the test utility (normally it is a perf problem, but that is not the case for test project)

```suggestion
            void Setup()
            {
                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
                {
                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: true);
                }
            };
```